typescript/sdk/src/providers/MultiProvider.ts

```diff
@@ -127,7 +147,7 @@ export class MultiProvider<MetaExt = {}> extends ChainMetadataManager<MetaExt> {
     this.providers[chainName] = provider;
     const signer = this.signers[chainName];
     if (signer && signer.provider) {
-      this.setSigner(chainName, signer.connect(provider));
+      this.setSigner(chainName, (signer as EthersSigner).connect(provider));
```

Comment:
dont really love this

Reply:
Yeah, here we force signer as EthersSigner because ZKSyncWallet requires ZKSyncProvider that has a couple of additional functions [mostly for bridging & reading some state from L1] we currently have no use for here, but the rest of funcs are pretty much same